hyperspy/io_plugins/attolight.py

```
@@ -0,0 +1,351 @@
+            if 'Resolution_Y' in line:
+                ny = int(line[line.find(':') + 1:-7])
+                # ny = pixel in y-direction
+            if 'Real Magnification' in line:
+                FOV = float(line[line.find(':') + 1:-1])
+            if 'Grating - Groove Density:' in line:
+                grating = float(line[line.find(':') + 1:-6])
+            if 'Central wavelength:' in line:
+                central_wavelength_nm = float(line[line.find(':') + 1:-4])
+            if 'Channels:' in line:
+                total_channels = int(line[line.find(':') + 1:-1])
+            if 'Signal Amplification:' in line:
+                amplification = int(line[line.find(':x') + 2:-1])
+            if 'Readout Rate (horizontal pixel shift):' in line:
+                readout_rate = int(line[line.find(':') + 1:-4])
+
+            if 'Exposure Time:' in line:
+                exposure_time_ccd_s = float(line[line.find(':') + 1:-3])
+            if 'HYP Dwelltime:' in line:
+                dwell_time_scan_s = float(line[line.find(':') + 1:-4]) / 1000
+            if 'Beam Energy:' in line:
+                beam_acc_voltage_kv = float(line[line.find(':') + 1:-3]) / 1000
+            if 'Gun Lens:' in line:
+                gun_lens_amps = float(line[line.find(':') + 1:-3])
+            if 'Objective Lens:' in line:
+                obj_lens_amps = float(line[line.find(':') + 1:-3])
+            if 'Aperture:' in line:
+                aperture_um = float(line[line.find(':') + 1:-4])
+            if 'Aperture Chamber Pressure:' in line:
+                chamber_pressure_torr = float(line[line.find(':') + 1:-6])
+            if 'Real Magnification:' in line:
+                real_magnification = float(line[line.find(':') + 1:-3])
+
+    # Correct channels to the actual value, accounting for binning. Get
+    # channels on the detector used (if channels not defined, then assume
+    # its 1024)
+    try:
+        total_channels
+    except:
```

Specify the error you expect to make sure you are not swallowing other bugs.

---

It's a `NameError`.